src/group_theory/perm/cycles.lean

```diff
@@ -617,6 +627,29 @@ begin
   { exact if_neg (mt same_cycle.apply_eq_self_iff (by tauto)) },
 end
 
+@[simp] lemma cycle_of_self_apply [fintype α] (f : perm α) (x : α) :
+  cycle_of f (f x) = cycle_of f x :=
+begin
+  rw (same_cycle.symm _).cycle_of_eq,
+  rw same_cycle_apply
+end
+
+@[simp] lemma cycle_of_self_apply_pow [fintype α] (f : perm α) (n : ℕ) (x : α) :
+  cycle_of f ((f ^ n) x) = cycle_of f x :=
+begin
+  rw same_cycle.cycle_of_eq,
+  rw same_cycle_pow_left_iff
+end
```

Comment:
The first `rw`s are really just `apply`, and in that case, why not just stay in term mode?
```suggestion
(same_cycle_apply.mpr (same_cycle.refl _ _)).symm.cycle_of_eq

@[simp] lemma cycle_of_self_apply_pow [fintype α] (f : perm α) (n : ℕ) (x : α) :
  cycle_of f ((f ^ n) x) = cycle_of f x :=
(same_cycle_pow_left_iff.mpr (same_cycle.refl _ _)).cycle_of_eq
```